cms/djangoapps/contentstore/tasks.py

```diff
@@ -1000,3 +1002,31 @@ def delete_v1_library(v1_library_key_string):
         "status": "SUCCESS",
         "msg": "SUCCESS"
     }
+
+
+@shared_task(time_limit=30)
+@set_code_owner_attribute
+def replace_all_library_source_blocks_ids_for_course(course, v1_to_v2_lib_map):
+    """Search a Modulestore for all library source blocks in a course by querying mongo.
+        replace all source_library_ids with the corresponding v2 value from the map
+    """
+    store = modulestore()
+    with store.bulk_operations(course.id):
+        for branch in [ModuleStoreEnum.BranchName.draft, ModuleStoreEnum.BranchName.published]:
```

Comment:
While we've agreed that v1 libraries will be purged to just two vesions each, the same doesn't hold for courses referencing libraries. Should we be concerned about leaving course versions in the database that continue to reference the old v1 libraries?